dwitter/templates/base.html

```diff
@@ -36,43 +36,50 @@
-      // Check for compatibility
-      if (window.localStorage) {
-        var switchTheme = function() {
-          // Shift themes
-          var theme = localStorage.getItem('theme');
-          if (!theme) {
-            localStorage.setItem('theme', 'dark');
-          } else if (theme === 'dark') {
-            localStorage.setItem('theme', 'green');
-          } else if (theme === 'green') {
-            localStorage.setItem('theme', 'hcontrast');
-          } else {
-            localStorage.setItem('theme', '');
-          }
-          location.reload();
-          // Call this function in the li to change the mode
+      (function() {
+        var themes = ['', 'dark', 'green', 'hcontrast'];
+        var themeUrls = {
+          dark: '{% static "theme_dark.css" %}',
+          green: '{% static "theme_green.css" %}',
+          hcontrast: '{% static "theme_hcontrast.css" %}'
         };
+        // Use saved theme if localStorage is available
+        var theme = window.localStorage ? localStorage.getItem('theme') : null;
+        // Ignore unknown theme
+        if (themes.indexOf(theme) == -1) {
+          theme = null;
+        }
+        // Use dark mode preference if no theme is set
+        if (typeof theme != 'string' && window.matchMedia) {
+          theme = matchMedia('(prefers-color-scheme: dark)').matches ? themes[1] : themes[0];
+        }
 
-        // Do the same thing outside the function without changing localStorage to set correct theme
-        var theme = localStorage.getItem('theme');
-        if (theme) {
-          var l = document.createElement('link');
-          if (theme === 'hcontrast') l.setAttribute('href', '{% static "theme_hcontrast.css" %}');
-          if (theme === 'green') l.setAttribute('href', '{% static "theme_green.css" %}');
-          if (theme === 'dark') l.setAttribute('href', '{% static "theme_dark.css" %}');
-          // if(theme==='blue') l.setAttribute('href', '{% static "theme_blue.css" %}');
-          l.setAttribute('type', 'text/css');
-          l.setAttribute('rel', 'stylesheet');
-          l.setAttribute('id', 'theme-style');
-          document.head.appendChild(l);
-        } else {
+        function updateTheme() {
           var l = document.getElementById('theme-style');
           if (l) {
             l.parentNode.removeChild(l);
           }
+          if (themeUrls[theme]) {
+            var l = document.createElement('link');
+            l.setAttribute('href', themeUrls[theme]);
+            l.setAttribute('type', 'text/css');
+            l.setAttribute('rel', 'stylesheet');
+            l.setAttribute('id', 'theme-style');
+            document.head.appendChild(l);
+          }
         }
-      }
+
+        updateTheme();
+
+        window.switchTheme = function() {
+          // Shift themes
+          theme = themes[(themes.indexOf(theme) + 1) % themes.length];
+          window.localStorage && localStorage.setItem('theme', theme);
```

Comment:
I moved the check inline because the theme can now be changed without `localStorage`.

Reply:
There is no `location.reload()` now. Does it successfully overwrite all previously loaded themes with the latest selected theme? Great if so!

(For that to work, I think every theme needs to override all properties of the other themes. Perhaps that is the case now.)

Reply:
It doesn’t have to overwrite them because their CSS is removed first. In all browsers I tested, that worked, ie. they didn’t hold on to removed CSS rules.